src/activation.js

```diff
@@ -180,11 +180,58 @@ function shouldContinue(output, router: Router) {
   return output;
 }
 
+//wraps a subscription, allowing unsubscribe calls even if
+//the first value comes synchronously
+class SafeSubscription {
+  //if this were TypeScript, subscriptionFunc would be of type
+  //(sub: SafeSubscription) => Subscription
```

Comment:
I haven't used Flow typings explicitly. Would it be the same type declaration as what TypeScript uses?